domain/schema/model.go

```diff
@@ -29,3 +30,17 @@ CREATE TABLE model_config (
 );
 `)
 }
+
+func spacesSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE spaces (
+    uuid            TEXT PRIMARY KEY,
+    name            TEXT NOT NULL,
+    is_public       BOOLEAN,
+    provider_id     TEXT -- provider_id is an empty string except for the providers that directly support spaces. At the moment, only MAAS (for which provider_id is "-1").
```

Comment:
I think we should use a foreign key table for actual types. Similar to https://github.com/juju/juju/blob/main/domain/schema/controller.go#L141-L152

Reply:
done